src/SqlAsyncCollector.cs

```diff
@@ -324,6 +328,18 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
             newDataQuery = $"WITH {CteName} AS ( SELECT * FROM OPENJSON({RowDataParameter}) WITH ({string.Join(",", bracketColumnDefinitionsFromItem)}) )";
         }
 
+        private static void ChangePropertiesToLowerCase(JObject jsonObject)
+        {
+            foreach (JProperty property in jsonObject.Properties().ToList())
+            {
+                if (property.Value.Type == JTokenType.Object)
```

Comment:
Seems better to actually check the C# type here (make sure it's actually a JObject)

Reply:
I didn't understand what you mean?

Reply:
`property.Value.GetType() == typeof(JObject)`

It seems odd to rely on Value.Type when you have the actual object - especially since we then cast it to JObject right below anyways.

Reply:
I was trying typeof(property.Value) == typeof(JObject) and it was showing me a compiler error, I need to learn when to use GetType vs typeof :|